torch/csrc/jit/autodiff.cpp

```diff
@@ -16,7 +16,7 @@ bool isDifferentiable(Node * n) {
   static std::unordered_set<Symbol> differentiable_kinds = {
     aten::add, aten::sub, aten::mul, prim::Constant, prim::ReplaceIfUndef,
     aten::sigmoid, aten::tanh, aten::mm, aten::chunk, aten::split, aten::t, aten::neg,
-    aten::unsqueeze, aten::expand, aten::addmm
+    aten::unsqueeze, aten::expand, aten::addmm, aten::gt, aten::lt, aten::eq, aten::ne, aten::ge, aten::le, aten::type_as
   };
```

Comment:
Can we make `type_as` non differentiable if the template tensor doesn't have a concrete tensor type? It shouldn't even attempt to do it if we don't have this information available statically.

Reply:
In many cases that would be too strict, as output of type_as likely won't require grad at all, so one does not care about information being available statically, and yet still have differentiable graph. isDifferentiable(Graph & g) should be smarter than it is now.

Reply:
Ok, I've pushed a commit, it's overly strict but will do for now.

Reply:
C2 test failures seem unrelated, MacOS failure is in jit, but I don't understand what it is and how the latest commit might have caused it.